doc/Data Plane Code Generation/Add_ProtocolMethods_In_HLCGenerated_SDK.md

```diff
@@ -0,0 +1,310 @@
+        await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+        var headers = new ServiceSetPropertiesHeaders(message.Response);
+        switch (message.Response.Status)
+        {
+            case 202:
+                return ResponseWithHeaders.FromValue(headers, message.Response);
+            default:
+                throw await ClientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
+        }
+    }
+
+    public ResponseWithHeaders<ServiceSetPropertiesHeaders> SetProperties(TableServiceProperties tableServiceProperties, int? timeout = null, CancellationToken cancellationToken = default)
+    {
+        if (tableServiceProperties == null)
+        {
+            throw new ArgumentNullException(nameof(tableServiceProperties));
+        }
+
+        using var message = CreateSetPropertiesRequest(tableServiceProperties, timeout);
+        _pipeline.Send(message, cancellationToken);
+        var headers = new ServiceSetPropertiesHeaders(message.Response);
+        switch (message.Response.Status)
+        {
+            case 202:
+                return ResponseWithHeaders.FromValue(headers, message.Response);
+            default:
+                throw ClientDiagnostics.CreateRequestFailedException(message.Response);
+        }
+    }
++
++   public virtual async Task<Response> SetPropertiesAsync(RequestContent content, int? timeout = null, RequestContext context = null)
++   {
++        Argument.AssertNotNull(content, nameof(content));
++
++        using var scope = ClientDiagnostics.CreateScope("ServiceClient.SetProperties");
++        scope.Start();
++        try
++        {
++            using HttpMessage message = CreateSetPropertiesRequest(content, timeout, context);
++            return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
++        }
++        catch (Exception e)
++        {
++            scope.Failed(e);
++            throw;
++        }
++    }
++
++    public virtual Response SetProperties(RequestContent content, int? timeout = null, RequestContext context = null)
++    {
++        Argument.AssertNotNull(content, nameof(content));
++
++        using var scope = ClientDiagnostics.CreateScope("ServiceClient.SetProperties");
++        scope.Start();
++        try
++        {
++            using HttpMessage message = CreateSetPropertiesRequest(content, timeout, context);
++            return _pipeline.ProcessMessage(message, context);
++        }
++        catch (Exception e)
++        {
++            scope.Failed(e);
++            throw;
++        }
++    }
+}
+```
+
+</details>
+&nbsp;
+
+**Note:** Protocol methods will be generated only in internal RestClient even if the `public-client` is set to true in the config. In order to add a public protocol method, you'd have to manually add it in public client and call the corresponding internal protocol method generated in RestClient.
```

Comment:
Could we call this out as a section at the same level as `## Add protocol methods in generated RestClient` ?

I think it'll be more clear if the reader can see it as two top-level steps:
1. Add methods to RestClient
2. Add methods to ConvenienceClient

We can provide samples for this section as well, just a simple call through from the Convenince Client to the RestClient should suffice.

Reply:
Updated!